src/SALib/analyze/morris.py

```diff
@@ -301,6 +312,7 @@ def _reorganize_output_matrix(
 
 def _compute_elementary_effects(
```

Comment:
Could you leave the original `_compute_elementary_effects()` function as it was and just pass in delta as a vector rather than a scalar?  The issue with the current modification is that you are changing the function call and the return type. Specifically, you add an extra argument and return a tuple of unscaled and scaled elementary effects you'll also have to rewrite all the tests for the function...

Alternatively, you could adjust the function to provide a scaled or unscaled calculation depending on whether `delta` is a scalar or a vector.

Or, a cleaner and less "magic" implementation could be to add a `scaled=False` argument.  Then you can test in the function:

```python
if scaled=True:
    # Do operations specific to scaled EE (e.g. compute delta)
else:
    # Do operations specific for unscaled EE

# Do operations that are common to both
```

Reply:
The delta was calculated based on the trajectory (giving a step per d_x). It needs reshaping and thus I think it makes more sense inside the function. So delta can stay a scalar but is unused in the scaled elementary effect case.